src/clusterfuzz/_internal/tests/core/bot/fuzzers/centipede/centipede_engine_test.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for centipede engine."""
+
+import os
+import shutil
+import unittest
+
+from clusterfuzz._internal.bot.fuzzers import engine_common
+from clusterfuzz._internal.bot.fuzzers.centipede import engine
+from clusterfuzz._internal.system import environment
+from clusterfuzz._internal.tests.test_libs import helpers as test_helpers
+from clusterfuzz._internal.tests.test_libs import test_utils
+
+TEST_PATH = os.path.abspath(os.path.dirname(__file__))
+DATA_DIR = os.path.join(TEST_PATH, 'test_data')
+WORK_DIR = os.path.join(DATA_DIR, 'workdir')
+CORPUS_DIR = os.path.join(DATA_DIR, 'corpus')
+CRASHES_DIR = os.path.join(WORK_DIR, 'crashes')
+
```

Comment:
Is this file copied at all from another file?

Reply:
It is based on the [`honggfuzz`'s test file](https://github.com/google/clusterfuzz/blob/c5d3476ef597addbfc8848b2c0ae1be1e428a55c/src/clusterfuzz/_internal/tests/core/bot/fuzzers/honggfuzz/honggfuzz_engine_test.py). I modified it to be suitable for `Centipede` but still tests similar criteria.
Is there any other criterion you'd like me to add?